flake.nix

```diff
@@ -13,77 +12,91 @@
   outputs = {
     self,
     nixpkgs,
-    flake-utils,
     rust-overlay,
     ...
   }: let
-    gitRev = self.rev or self.dirtyRev or null;
-  in
-    flake-utils.lib.eachDefaultSystem (system: let
-      pkgs = import nixpkgs {
-        inherit system;
+    inherit (nixpkgs) lib;
+    systems = [
+      "x86_64-linux"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+    ];
+    eachSystem = lib.genAttrs systems;
+    pkgsFor = eachSystem (system:
+      import nixpkgs {
+        localSystem.system = system;
         overlays = [(import rust-overlay)];
-      };
-
-      # Get Helix's MSRV toolchain to build with by default.
-      msrvToolchain = pkgs.pkgsBuildHost.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
-      msrvPlatform = pkgs.makeRustPlatform {
-        cargo = msrvToolchain;
-        rustc = msrvToolchain;
-      };
-    in {
-      packages = rec {
-        helix = pkgs.callPackage ./default.nix {inherit gitRev;};
-
+      });
+    gitRev = self.rev or self.dirtyRev or null;
+  in {
+    packages =
+      lib.mapAttrs (system: _: {
```

Comment:
Interesting, why did you choose to use `mapAttrs` here instead of using `eachSystem`, since you're not accessing the `pkgs` value? You opted to use `inherit (pkgsFor.${system})` for the sake of being explicit, but still have an occurrence of `pkgsFor` on the outside.

Does it make more sense to you this way? I'm very curious why if that is the case.

Reply:
That applies to using the `pkgs` argument too 😄